src/index.js

```diff
@@ -41,7 +45,30 @@ export const createInstance = (options = {}) => {
     createNamespacedStorage,
     getMonitors,
   });
-  const instance = createExecuteCommand({ instanceName: name, logController });
+
+  const instance = createExecuteCommand({
+    instanceName: name,
+    logController,
+    components,
+  });
   logController.logger.logOnInstanceCreated({ instance });
+
   return instance;
 };
+
+export const createInstance = (options = {}) => {
+  const eventOptionsValidator = objectOf({
+    name: string().default("alloy"),
+    monitors: arrayOf(objectOf({})).default([]),
+  }).noUnknownFields();
+
+  const { name, monitors } = eventOptionsValidator(options);
+
+  return createCustomInstance({
+    name,
+    monitors,
+    components: optionalComponents,
+  });
+};
+
+export { optionalComponents as components };
```

Comment:
If somebody wants to use alloy in their NPM project they can add something like the following code in their main.js file (tree shake will work in this case):

```
import { components, createCustomInstance } from "@adobe/alloy";

const alloy = createCustomInstance({
  name: "alloy",
  components: {
    privacy: components.privacy,
    personalization: components.personalization,
  },
});

alloy("configure", {
  datastreamId: "bc1a10...",
  orgId: "5B....@AdobeOrg",
  debugEnabled: true,
});
```
Tree shake doesn't function in all the case. For example, tree shake will not function in the following code:

```
import { components, createCustomInstance } from "@adobe/alloy";

const { privacy, personalization } = components; // This is the line that would break the tree shake.

const alloy = createCustomInstance({
  name: "alloy",
  components: {
    privacy,
    personalization
  },
});

alloy("configure", {
  datastreamId: "bc1a10...",
  orgId: "5B....@AdobeOrg",
  debugEnabled: true,
});
```

Reply:
It seems a more natural API is passing in an array. I see why you accept an object though because you can just "components: optionalComponents", but you could just do "components: Object.values(optionalComponents)".